auto_route_generator/lib/auto_route_generator.dart

```diff
@@ -72,8 +72,8 @@ class AutoRouteGenerator extends Generator {
       '.gr.dart',
     );
 
-    final hasPart = clazz.library.parts.any(
-      (e) => e.uri?.endsWith(part) ?? false,
+    final hasPart = clazz.library.parts2.any(
```

Comment:
```suggestion
    final hasPart = clazz.library.parts2
        .map((e) => e.uri)
        .whereType<DirectiveUriWithRelativeUriString>()
        .any((e) => e.relativeUriString.endsWith(part));
```

Reply:
Hey Josh, thanks for the review and suggestions!
I believe both of those implementations will lead to the same outcome. As someone with no experience in `analyzer` I find it hard to make a weighted judgment whether to go with a simpler or a more robust implementation and will defer the decision to package maintainers. I am happy to support both.
I believe repository maintainers should be able to make changes to this PR, and thus accept these suggestions if they better fit the needs of the package.